service/worker/pernamespaceworker.go

```diff
@@ -547,6 +571,7 @@ func (w *perNamespaceWorker) stopWorkerAndResetTimer() {
 
 	w.stopWorkerLocked()
 	w.retrier.Reset()
+	w.reserved = false
```

Comment:
Is this the only place we need to clear the reserved flag?
I'm not super familiar with this code, which is quite complex. Could you please explain in a comment why this is done here and not in `stopWorkerLocked` for example?

Reply:
It could be done there too, there's not a really strong argument for either one, I think. The difference is whether it takes an additional rate limiter token on retry or not. Given that no rate limit at all mostly works fine, I don't think it matters much. I picked "no" since I figure if Start fails, then it hasn't done any polls yet.

I'm not sure what a comment should say.. just that bit about retries?

Reply:
Yup, just that bit about retries. That's the piece I was missing and I think putting that in the code has value for future readers.